Doc/library/io.rst

```diff
@@ -106,6 +106,54 @@ stream by opening a file in binary mode with buffering disabled::
 The raw stream API is described in detail in the docs of :class:`RawIOBase`.
 
 
+.. _io-text-encoding:
+
+Text Encoding
+-------------
+
+The default encoding of :class:`TextIOWrapper` and :func:`open` is
+locale-specific (:func:`locale.getpreferredencoding(False) <locale.getpreferredencoding>`).
+
+But many developers forget to specify encoding when opening text files
+encoded in UTF-8 (e.g. JSON, TOML, Markdown, etc...) since most Unix
+platforms uses UTF-8 locale by default. It cause a bug because locale-specific
+encoding is not UTF-8 for most Windows users. For example::
+
+   # may not work on Windows when non-ASCII characters in the file.
+   with open("README.md") as f:
+       long_description = f.read()
+
+Additionally, Python may change the default text encoding to UTF-8 in the
+future, although there is no plan yet.
```

Comment:
```suggestion
Additionally, while there is no concrete plan as of yet, Python may change
the default text file encoding to UTF-8 in the future.
```

Use clearer and more idiomatic phrasing and diction, and specify that this is the "file" encoding, as opposed to the many other default encodings (console, filesystem, source, etc)